[WIP] Game Mode - Natural Disaster

RaveRadbury opened the pull request:
All the basic mechanics are in, all the timings need adjustment. AT LEAST. There's probably more to fix.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Space Station 13 is a game of memes, reference, and trope. Here's one that we don't currently cover: The Disaster Movie. Band together as a crew to salvage from the station and tame the wilds!

Disaster is a game mode that forces the crew to evacuate the station due to intense and frequent waves of catastrophic weather. It ends when all the megafauna are dead so NT can perform a safe extraction, or when the number of surviving crew drops below a certain percentage (currently set to 25%, but I'd like to see if 0% could work).

This takes inspiration from modes like Meteors and Goon's Build Your Own Station. However the events don't directly destoy the station and there are no starting materials spawned. I hope for this to have a bit of a Rimworld or Dwarf Fortress vibe as engineering expands the mining camp with materials brought in by the miners.

This also puts the crew in more direct conflict with the Ashwalkers, who can achieve a Crew Loss by contributing to the deaths of spessmen. They might require some rebalancing for this mode.

Aside from the Ashwalkers there are no other intended antagonists. I think that this can function as the randomized PvE mode that people are looking for.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This mode seeks to address a number of issues.
 - People want a new game mode
   - but they DON'T want it to be conversion
   - AND they want to keep doing their jobs
 - There's also been talk of mining being too seperate from everyone else, NOT ANYMORE
  - This also encourages early prep and staging of the Aux Base, as it allows a more public shuttling down to Lavaland.
 - Some jobs are essentially unchanged, 
   -  meanwhile other jobs like engineering and atmos get to actually create infrastructure. 
   -  As the mode is now, Supply will likely amass protective clothing so they can salvage the station and make use of the (thankfully close by) cargo shuttle.
 - For once Red Alert launching to Lavaland actually makes sense and has an impact on early and mid game.
 - Ashwalker respawn mechanics allow few or many ghosts to play an antag role. (Likely to pick up speed as the deathcount increases and people give up on being revived)
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Game Mode: Disaster
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
TODO for playtest
 - [x] Events begin occuring after 10 minutes
 - [x] Events are called every 2-2:30 minutes
 - [x] Only events unique to this game mode are called (probably a better way to do this)
 - [x] Need to keep the shuttle from being called
 - [x] [Provide documentation on the wiki for the game mode so that everyone can know what's up.](https://tgstation13.org/wiki/User:RaveRadbury/Disaster)
 - [x] Make sure that Crew Win procs round end
 - [x] Make sure that Crew Loss procs round end

TODO for final
 - [ ] The telegraph of the weather probably needs to be tweaked
 - [ ] Investigate why the hell cold weather is circumvented by removing hardsuit
 - [ ] Just use the weather system instead of piggybacking the events system


Thanks to @JJRcop  for crucial debugging assistance.

RaveRadbury committed:
Natural Disaster

All the basic mechanics are in, all the timings need adjustment. AT LEAST. There's probably more to fix.

RaveRadbury commented:
Well, Travis is upset with me, guess I'll have to find out why tomorrow.

Borisvanmemes commented:
So uhh, how does it work?

RaveRadbury commented:
> So uhh, how does it work?

I took all 5 weather types we have and created subvariants that can penetrate the entire station with only shuttles left unscathed (for late joiners). They are going to be going off randomly for the rest of the round. 

If all the megafauna are dead, the round is over and the crew wins. 
If most of the crew dies, it's a crew loss.

Whoneedspacee commented:
hey guys its meteors except weather now haha

JJRcop commented on code/game/gamemodes/disaster/disaster.dm:
```diff
@@ -0,0 +1,72 @@
+/datum/game_mode/disaster/generate_report()
+	return "Several disasters are incoming. Evacuate to Lavaland and see to it that work continues. We can't lose money over a little space-weather! While you're at it, if you could kill of the most dangerous wildlife down there, we could arrange for your extraction. Just something to think about."
+
+/datum/game_mode/disaster/announced
```
This makes two gamemodes, rather than defining the properties after one is announced. You probably got this from the code for extended. In that case, it was intended to have two gamemodes: one that's secret (`secret_extended`) and one that's not secret, announced to be extended (`extended`). You also have put the config_tags as the same which will clash, which makes me believe you do NOT want two separate gamemodes.

You should remove the block from 20 - 23 and remove `/announced` from the rest of these procs

JJRcop commented on code/game/gamemodes/disaster/disaster.dm:
```diff
@@ -0,0 +1,72 @@
+		SSticker.mode_result = "loss - Massacred by Nature"
+		SSticker.news_report = DISASTER_LOSE
+
+/datum/game_mode/revolution/special_report()
```
revolution

JJRcop commented on code/game/gamemodes/disaster/disaster.dm:
```diff
@@ -0,0 +1,72 @@
+	config_tag = "disaster"
+	false_report_weight = 0
+
+/datum/game_mode/disaster/announced/generate_station_goals()
```
This was copypasted from extended and should be removed.

JJRcop commented on code/datums/weather/weather_types/floor_is_lava.dm:
```diff
@@ -38,3 +38,7 @@
 		return
 	L.adjustFireLoss(3)
+
+datum/weather/floor_is_lava/molten
```
Include leading slashes on a few of these

JJRcop commented:
I'm too tired to go through it more and there's a lot more that someone who knows gamemode code better would be able to say. Normally I would research the gamemode chain to find out other details about this but I need to sleep.

Borisvanmemes commented:
On one hand that sounds kinda boring and a pain in the ass to play with, on the other hand that sounds highly unique and i wanna try it tbh.

subject217 commented:
we should probably reserve the remarks for how it plays out once it actually gets playtested

knacker48 commented:
Well atleast its not hivemind

habader commented:
Think you could force the golem, ashwalker and lifebringer ruins to always spawn on this gamemode?

knacker48 commented:
> Think you could force the golem, ashwalker and lifebringer ruins to always spawn on this gamemode?

I kind of get forcing ashwalkers as a guaranteed antagonist, but whats the point with golems and lifbringers?

habader commented:
> > Think you could force the golem, ashwalker and lifebringer ruins to always spawn on this gamemode?
> 
> I kind of get forcing ashwalkers as a guaranteed antagonist, but whats the point with golems and lifbringers?

Golems could lead to some interesting interactions and lifebringers are pretty cool

RaveRadbury committed:
fix

hehe, where did this revolution come from? It's a disaster here.

RaveRadbury committed:
cleanup

removed unnecessary copypasta

RaveRadbury committed:
leading slash

RaveRadbury committed:
leading slash

RaveRadbury committed:
slashy

RaveRadbury commented:
@Whoneedspacee I didn't even look at the Meteor game mode, what an absolute fool I am. If I had, my code would look a lot better!

necromanceranne commented:
Forcing the lavaland spawns forces varying factions in the game to create that necessary conflict. I might even recommend having some players start the game mode AS these factions with additional equipment to expand and present a credible threat/rivalling faction. Beach bums for life.

I'd maybe even recommend adding the ability for the various midround antags to spawn on lavaland/navigate to lavaland during this mode. That said, some might be stronger than others (nightmares have an incredibly easy time on lavaland, while abductors would struggle massively). A few should probably be disabled during this same mode (blob, lone op), since they're impossible to complete or have easy victories.

JJRcop commented on code/game/gamemodes/disaster/disaster.dm:
```diff
@@ -0,0 +1,58 @@
+var/finished = 0
+
+/datum/game_mode/disaster/pre_setup()
+	SSevents.scheduled = 3000
```
This should be `world.time + 5 MINUTES`

JJRcop replied:
the reason this didn't work out is because world.time ticks the entire time the server is open, not just when the round starts

kriskog commented:
Sounds promising, for a fairly rare mode. What concerns me is quite a few jobs require a fair amount of stuff to function. Maybe start the round out in the middle of evac, having already prepped some crates with essential machinery/stuff from departments to avoid too much repetitive deconstruction every time.

PKPenguin321 commented:
It's like the old WHO DIE RAST event but without the gameshow theme

RaveRadbury committed:
Time fix

ATH1909 commented:
I feel like this should be an admin spawn-only gamemode, since the novelty of it could wear off after a while.

If the ashwalkers are supposed to be the intended antagonists, I'd suggest ensuring that (only on disaster rounds) they'll always spawn and (maybe) have the ability to "convert" a lavaland spawner tendril into a new ashwalker tendril if their original tendril gets destroyed somehow.

blessedmulligan commented:
> 
> 
> I feel like this should be an admin spawn-only gamemode, since the novelty of it could wear off after a while.
> 

I feel like this would be similar to blob in that it's an "oh shit drop everything" gamemode, but it's a lot less linear than blob, has more potential for long rounds, and has more variety in things to do. Also, it lets people make use of mechanics that don't get used too often.

anconfuzedrock commented:
The entire fucking day and I (think) I understand now that the intent is the crew fucking off to lavaland to kill megafauna, which is arbitrary as fuck.

One of the reasons lavaland is shit is its mechanics are completely at odds with everything else in the game, and so this will basically just turn into "60 miners vs megafauna" since kinetic accelerators are the only good weapon. Lasers are fucking shit against lavaland monsters already, half damage just makes it even worse.  It's difficult for me to understate just how much more powerful and efficient a kinetic accelerator is than anything on station, how useless a hardsuit or any protection is with the slowdown it causes, How much more unbalanced it is with easy access to luck based katanas, holoparasite, actual self antagging (wizard is one dumb thing but fuck morphs).
As such, going to lavaland to fight fauna will simply devolve into 60 people fighting over the kinetic accelerators since everything else is useless (a fucking security officer is worse equipped than a botanist on this one, botanist at least has a hatchet) and I'd much rather see our crappy gamemodes touched up first before we add more crappy ones.

Cyprex commented:
While the premise sounds interesting, how does this interact with our various maps?

RaveRadbury committed:
Fixed Travis issue

RaveRadbury committed:
That made it worse, reverting.

RaveRadbury committed:
Missed some _ should pass checks now.

RaveRadbury commented:
@Cyprex can you narrow that question down a little? I'm not really sure what to say,

kevinz000 commented:
planetstation

Grazyn commented:
One thing's for sure, there won't be any shortage of ghosts. With weather permeating the entire station including maint, anyone caught too far from the shuttles will die, period. There's no "well maybe I can hold a bit longer to do my thing", you either go to lavaland or you die.

Cyprex commented:
Does this require a specific map, or modify the normal maps in any way?

plapatin commented:
stop trying to make meteors good

FloranOtten commented:
I would advise having an NT drop pod or something of building materials. If you want people to continue their jobs I would hope they wouldn't have to wait for science to research their department. We'll need extra RCDs and RPDs if you don't want it to just be a bunch of greyshirts waiting for the few engis to finish the station. 

Protolathe boards because the heads will never take them with them

Blueprints so a CE-less station isn't fucked

And I would advise having a large ship dock at whiteship dock or a small shuttle dock at the transit pod dock because on highpop it's just going to be herding cats. 

I would advise having higher engineering slots because we're going to need it

For this we need a power source that'll work on lavaland. Singulo doesn't work. Tesla tends to escape. I've seen SM designs that dump the waste into a large airthight room and abusing lavalands floors to delete it but it shouldn't be relied upon.

We need to get comms boards because people will abandon station power, comms will lose power and now you don't have comms on lavaland.

Metal. Holy fuck, metal. I doubt all the metal on lavaland and on station is enough to build an entire station, including furnishing.

Who is the antag here? Who is opposing the crew? Megafauna, sure, but that's a side antag. It's there but it won't harm you if you don't harm it.

All in all, good idea, many flaws. It's got every flaw that planetstation has, and more.

RaveRadbury commented:
> Does this require a specific map, or modify the normal maps in any way?

Nope! This game mode is meant to pre-existing maps seem fresher as the player has to evaluate the kinds of things they want to evacuate with, the things that they'll need scavenged later, and find optimal paths for a quick evacuation.

RaveRadbury committed:
Roundstart events triggers properly

RaveRadbury committed:
Weather events start correctly.

JJRcop commented on code/datums/weather/weather_types/floor_is_lava.dm:
```diff
@@ -38,3 +38,7 @@
 	L.adjustFireLoss(3)
+
+/datum/weather/floor_is_lava/molten
+			target_trait = ZTRAIT_STATION
```
too much indentation

RaveRadbury committed:
indentation

RaveRadbury committed:
Tweaks

The blacklist thing is something I'm talking out with people atm but if anyone is going to test this, they should do it with the blacklist downloaded.

RaveRadbury committed:
Merge branch 'Disaster' of https://github.com/RaveRadbury/tgstation into Disaster

MrDoomBringer commented:
Would be interesting if this gamemode made the megafauna 'bloodlusted' or something that made their AI slowly wander them towards the lavaland base. (Or maybe spawn special new megafauna that did this). Just in order to keep the crew involved and invested in the gamemode, rather than "sit around in lavaland and wait for the miners to go win the game for us"

HomocidlePancakes commented:
*looks over at colossus in close proximity of mining base*

Whoneedspacee commented:
Might want to stick some VR sleepers on the mining map as well. Having megafauna training up for people to play while being stuck on the mining base would be neat.

RaveRadbury commented:
[I'm linking the documentation I have for the mode so far here](https://tgstation13.org/wiki/User:RaveRadbury/Disaster) and on my checklist. Hopefully this will provide the information and guidance so that people don't immediately die through lack of knowledge.

RaveRadbury committed:
Prevents the shuttle from being called

Lobachevskiy commented:
https://www.youtube.com/watch?v=rdvymduAbtc

Tlaltecuhtli commented:
idk it can already be done by announced extended + people are bored enough to go to lavaland

main problems with building anything on lavaland  takes sooooooooo long without making things look like shit

Lobachevskiy commented on code/game/gamemodes/disaster/disaster.dm:
```diff
@@ -16,6 +16,9 @@
+	SSshuttle.registerHostileEnvironment(src)
+
 /datum/game_mode/disaster/generate_report()
 	return "Several disasters are incoming. Evacuate to Lavaland and see to it that work continues. We can't lose money over a little space-weather! While you're at it, if you could kill of the most dangerous wildlife down there, we could arrange for your extraction. Just something to think about."
```
"if you could kill **of the most** dangerous wildlife down there"

RaveRadbury committed:
Fixed a typo

Swiftkor commented:
I remember a station evacuation event that the admins did a while ago, it was one of the best rounds I've had and sadly ended too quickly.
I agree that engineering should get some extra resources to build the station, maybe nanotrasen airdrops materials and some ripleys with RCDs. They could also drop some sort of gas miner device so atmos techs could put their big brains to use and make functioning atmos systems. Some sort of blueprint projector device for the CE could be useful as it would allow the CE to make a basic outline for a station so that it doesn't look like complete shit.

RaveRadbury commented:
It seems like the wincons aren't proc'ing. Once I figure that out I think it will be at the point where a testmerge would be justified.

Ordonis commented:
Gamemode does not automatically play Run (I'm a Natural Disaster) on repeat at 150dB until the crew wins, terrible oversight. In all honesty though, this seems like a good gamemode that avoids the pitfalls of a lot of the non-traitor game ending modes like Flops or Cult/Reeeevs, in that you can still do your job to an extent.

RaveRadbury committed:
creates a win_check proc for megafauna death

RaveRadbury committed:
Merge branch 'Disaster' of https://github.com/RaveRadbury/tgstation into Disaster

wesoda25 commented:
How is this any different from those admin events when you build a new station on lavaland? Those are run every now and then, but good lord would we get burnt out fast. Is there anyway you could survive on the station? Sounds to me if you’re anywhere on the station you get hit by acid rain and thats game. I simply don’t understand what makes this an actual game mode. It definitely appears as if it would need to be admin only. I also don’t understand what megafauna have to do with the gamemode whatsoever. Are they cursing the station or something? 

It feels to me that this gamemode is trying to be as far removed from space as possible from a space oriented game. Either I don’t understand large parts of the explanation, or the gamemode itself doesn’t make sense.

Whoneedspacee commented:
i mean it's pretty much the build your own station gamemode. i personally don't understand why the megafauna part exists though because i personally would probably just roundstart rush every megafauna because why not.

it seems like this gamemode needs a unique pve antagonist if it wants to be anything special (i'm not sure it wants to be special).

still though compared to some gamemodes i'm sure some people would love this just because it's different so i see no reason to at least not have it, even if on a really low rotation probability.

RaveRadbury commented:
@wesoda25 bad weather forces you to evacuate. In order to be rescued the are needs to be secured, therefore Lavaland needs to be free of megafauna so centcomm can perform an extraction.

wesoda25 commented:
I guess that makes sense. I do like the idea of sending salvage missions up to the station in between storms and what not. I just think people will get wary after a time, not to mention start rushing fauna like space said.

Dennok commented:
You, or someone, can make bluspace shelter capsules for RnD, Brig, Brige, Engi, Bar, Chappel, Botany, and all what can be need to the crew, and send it as CentCom help, or add it to the cargo. So lost and hungry (lazy) spacemans can easy make base on Lavalend.

RaveRadbury committed:
Added wincon checks, updated lavaruinblacklist.txt

It's trashy to mangle the config like this, but I don't get mapping well enough yet to force certain ruins dependent on my game mode.

RaveRadbury commented:
This is ready for testmerge. I'd like to see how the rounds play out with the event blacklist situation along with the lavaland config.

Both of these things will need to be removed before this can be ready for a full merge. There are probably _some_ events that can work with this game mode. I'll also need to find a way to force the faction ruins (provided that these initial tests go well), likely ashwalkers at the least.

RaveRadbury commented:
New bugs from the testmerge
 - Players couldn't build rooms
 - Mid-Round Antags kept getting called
 - Ruins didn't spawn the way I planned
   - Editing the config was a dumb idea and I need to actually force the ashwalker ruin (maybe lifegivers, maybe golems) if the mode is selected.

RaveRadbury committed:
Reverts lavaland blacklist, forces Ashwalkers, Golems, and Lifebringers, blacklists each standard event for disaster

RaveRadbury commented:
- I've successfully built rooms on public and private servers. Worthy of note is that rooms cannot be added onto the aux base. This may have been what caused the issues
  - I've blacklisted disaster on each event so hopefully this will stop mid-round antags
  - I changed Ashwalkers Lifegivers and Golems to force spawn rather than fiddling with the blacklist like a dummy.

Should be ready for another test merge.

RaveRadbury committed:
Adds BIG ANNOUNCEMENTS to weather events, gives unique flavor text.

Pandemonium1778 commented:
Love the concept, though I think forcing the removal of the megafauna to be the criteria for crew victory to be a bit much. There's not enough Kinetic Accelerators for this sort of thing and as others have mentioned, it basically turns into 40 - 60 people swarming lavaland to wipe out megafauna.

I think it'd be better suited to turn this gamemode towards more of a PvP angle with a focus on ashliggers vs station crew, ashliggers having to wipe out the crew to win and the crew having to destroy something in the ashligger nest to allow for centcomm to come and extract them. Something like that sounds like a better approach to me.

Timonkeyn commented:
if this happens at any time im just gonna fuck off to space because this sounds boring as shit to just have the same 5 weathers over and over again and basically rushing all over the Station to stay alive

Whoneedspacee commented:
pretty sure thats the point of the gamemode dude

RaveRadbury committed:
Removes lifebringers and golems from forced spawn. Blacklists some more events.

A bunch of antag events are bundled with their antag stuff and should be moved in a later PR

RaveRadbury commented:
- I think I've finally blacklisted every event so I can finally get a controlled test of that. 
- Weather has full on priority announcements. That might be terrible, we will see. 
- Only ashwalkers are force spawned now

Arianya commented:
Currently testmerged on Bagil

optimumtact commented:
Anyone else notice the PR ends in 666 ?

optimumtact commented on code/datums/weather/weather_types/radiation_storm.dm:
```diff
@@ -45,8 +45,7 @@
 		L.rad_act(20)
 
 /datum/weather/rad_storm/end()
-	if(..())
```
why are you changing the behaviour of the parent return ?

RaveRadbury replied:
Coderbus advised I do this as the change I made _shouldn't_ make any actual changes? It seems like some funky code. Am I breaking something here?

optimumtact commented on code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:
```diff
@@ -63,6 +63,8 @@
 
 /mob/living/simple_animal/hostile/megafauna/death(gibbed, var/list/force_grant)
+	if(SSticker.mode)
+		SSticker.mode.check_win() //Calls the rounds wincheck, to see if all megafauna have died
```
you shouldn't call it for every single mode needlessly

optimumtact replied:
this should be a signal

optimumtact replied:
then your gamemode can register for the signal from each of the megafauna at roundstart

Whoneedspacee replied:
pretty sure this is how it’s checked for other gamemodes but i think there’s also another proc used in ops that doesn’t involve shoving these around

optimumtact replied:
that DOES NOT MAKE IT RIGHT

Whoneedspacee replied:
literally how every other gamemode does it so

optimumtact replied:
you can keep repeating how bad the rest of them are at doing it, that won't change how it should be done

Whoneedspacee replied:
im telling you that he shouldn't be putting the check there in the first place. there's talk of phasing out check_win in the first place.

if he does something like locate(megafauna_type) in glob.megafauna_mobs or something in check_finished then why not

optimumtact commented:
also right now the maploading happens before round selection, so you'd probably need to tweak a whole bunch of things if you wanted ruins loading based on round type

Whoneedspacee commented:
or you could just paste map templates at specific points in the map based on waypoints like the ones used in blob, that’s probably easier

zombiegums commented:
I  think this game mode is really fun and unique. I like that it encourages the crew to collaborate in order to survive. As a borg player, it's also nice to be able to hop between topside and Lavaland and scavenge for things/bring down bodies of players who were unfortunate enough to get caught up in a storm. I found that there was always something for me to do, especially as an engiborg.

ArcaneMusic commented:
So, thoughts from observing the test round that was played earlier today. First off, weather wasn't very impactful to the station. Things like fire/ice/acid raining down on the station doesn't cause any real structural damage, or limit the crew's ability to use resources on station effectively. In addition, the issue seems to be that on lavaland, most challenges are... optional. You don't NEED to leave the lavaland base to subsist off of it, as outside of ashlizards (Barely) and the Syndicate Comms Base (Which shouldn't be attacking mining in the first place, just check my notes), there aren't really any non-mob threats on lavaland.  So, ideas to make the whole "Lavaland is out new home" shtick really kick in harder!

-Firestorms on station could create localized pyroclastic anomalies on station. Some number ramping up from 1-6 in addition to the ashstorm effect, so that there's a chance that critical areas of the station may be consumed by fire. As it stands right now, there are no consequences to very flammable areas on station like medical storage or toxins being caught in a firestorm.
-There's some room to really work on the _**COLD**_ infrastructure. As it stands,  snowstorms could have the chance to build/produce snow tiles/walls against airlocks, or to freeze up air pipes and cause them to burst. Suddenly getting buried into mait sounds like a VERY natural disaster thing that could happen.
-I like what you mentioned acid should do, I'm just not fully sure if acid was actually being applied to objects that wern't worn or held. Thoughts?
-Potential idea: A weaker form of weather that could maybe only occur in the first 10-15 minutes, which would produce small pockets of water vapor "Drizzle" across the station, slowing down crew without slip resistant gear like galoshes.

wesoda25 commented:
Can this be untestmerged until its not a WIP? From what I can tell there are numerous flaws, and most of the players don't like it in its current state. I feel multiple aspects of the game are unfinished and rushed, and it leaves a very boring round thats hardly different than a common admin event. In the feedback thread you said the worse possible scenario is a slow round culminating in nothing until everyone is finally dead, and 3/3 rounds I've observed have all ended this way. Each time some sort of disaster has been required, may it be admins spawning a wizard, or ash lizards kiting fauna to the station. 

As for improvement, I've got this:

- Storms should actually destroy the station

- Arrivals need some way of not instantly being removed from the round

- The goal of the gamemode should not be to kill megafauna

RaveRadbury commented:
> Can this be untestmerged until its not a WIP? 
The mode can only be run by an admin manually selecting it, so I don't really understand why you want it removed. Does it's mere existence cause you pain?

ArcaneMusic commented:
> > Can this be untestmerged until its not a WIP?
>  The mode can only be run by an admin manually selecting it, so I don't really understand why you want it removed. Does it's mere existence cause you pain?

It's not that like it's cancerous to the touch, and the mere mention of the word "disaster mode" causes fear in a man's eyes, but it's been tested a few times now, and the overwhelming consensus after each round so far has been that it's really not ready to take out of the oven to test again until it gets more features.

RaveRadbury commented:
>Overwhelming consensus

>Most players

Where are you guys even getting these numbers from? I've been doing daily builds for the last three days on this and it's not like I can test some of the more complicated balance aspects of a game mode on a private server (especially with my current internet service).

I've done fewer than 10 tests of this mode, so unless admins are running it when I'm not around I find it hard to believe that removing the testmerges is even something to talk about at this point. 

Having said all of this, it actually doesn't matter much to me whether the testmerges get removed or not, as I will just get another testmerge later when I need it. However, I would appreciate it if you guys would contact the keyholders on discord instead of doing this here.

optimumtact commented:
>overwhelming consensus

I believe this

>Most players

Me

handy translation guide for tg repo

OnlineGirlfriend commented:
After playing three test rounds, I enjoy this game mode greatly and look forward to its realization.

ATH1909 commented:
Yo, what if we combined this gamemode with the old away missions? Like, have a chance to make the place that everyone needs to be evacuated to be an away mission (accessed through the gateway?) instead of lavaland (which would have a permanent ash storm or something if it wasn't chosen as the place to be evacuated to).

2whatever2 commented:
When i played this gamemode as a late joiner it was horrible to deal with as i was immediately assaulted by a acid rain and snow disaster while im trying to pull up the wiki for this gamemode, the second time i played this as an AI no one listened to eachother and the shuttle was called immediately.

All i have to say is this gamemode was pretty fun when it started but when i realized we are going to do this for 30+ more minutes and half the station was dead because no one communicated i dont really like this gamemode as much anymore because you require team effort and as much as i want to do my job still while im bringing the protolathe and autolathe boards and other circuit boards, Why do i have to deal with the people who drop them into lava and bring no circuits with them at all?

adamkad1 commented:
> When i played this gamemode as a late joiner it was horrible to deal with as i was immediately assaulted by a acid rain and snow disaster while im trying to pull up the wiki for this gamemode, the second time i played this as an AI no one listened to eachother and the shuttle was called immediately.
> 
> All i have to say is this gamemode was pretty fun when it started but when i realized we are going to do this for 30+ more minutes and half the station was dead because no one communicated i dont really like this gamemode as much anymore because you require team effort and as much as i want to do my job still while im bringing the protolathe and autolathe boards and other circuit boards, Why do i have to deal with the people who drop them into lava and bring no circuits with them at all?

isnt the arrivals shuttle considered a shuttle thus immunue to hazards? you should stay inside it when wiki'ing

RaveRadbury commented:
Been moving over the last week, hence the lack of activity.

Current plan is to:
  * Address Orange review
  * Integrate the new weather into the actual weather system
  * Roll back some of the event blacklist, allowing for some side antagonists
  * For now I'm abandoning forcing the ashwalker spawn. I think that @Whoneedspacee's idea for placing a ruin in a way similar to the bluespace shelters is the best way to go about it.

 With these changes it will be a basic functional mode and I think that's an alright start. If I go any further than this I'm likely to be so far out of my depth that I wouldn't be able to finish.

RaveRadbury committed:
Allows side antags and events, Removes forced Ashwalker spawn, New Siren

Condenses weather deciseconds using time defines. Removes some unecessary telegraph_duration declarations. Nerfs acid, adjusts timings and probabilities of whitelisted weather events.

RaveRadbury commented:
I need to deal with some IRL stuff right now and I don't want this to go stale so I'm gonna close it for a bit.

Tharcoonvagh commented:
For a bit, he said
By fall, he said

boodaliboo commented:
> For a bit, he said
> By fall, he said

rip this pr
